From e6780a1d661ab2a76c16602edfc3ca08e97612bf Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Tue, 21 Sep 2021 13:12:09 -0700
Subject: [PATCH] Fix additional compiler warnings

Fixes
../../base/files/file_util_linux.cc:33:10: error: case value evaluates to -1859950530, which cannot be narrowed to type 'unsigned long' [-Wc++11-narrowing]                                                         case static_cast<int>(BTRFS_SUPER_MAGIC):
         ^

Jan 2023:
  file_util_linux.cc was removed but sys_info_posix.cc remains so patch adjusted.

Upstream-Status: Pending
Signed-off-by: Khem Raj <raj.khem@gmail.com>
Signed-off-by: Randy MacLeod <randy.macleod@windriver.com>

---
 base/system/sys_info_posix.cc | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/base/system/sys_info_posix.cc b/base/system/sys_info_posix.cc
index dbf9947723..921d31464c 100644
--- a/base/system/sys_info_posix.cc
+++ b/base/system/sys_info_posix.cc
@@ -64,8 +64,8 @@ bool IsStatsZeroIfUnlimited(const base::FilePath& path) {
 
   switch (stats.f_type) {
     case TMPFS_MAGIC:
-    case static_cast<int>(HUGETLBFS_MAGIC):
-    case static_cast<int>(RAMFS_MAGIC):
+    case static_cast<unsigned int>(HUGETLBFS_MAGIC):
+    case static_cast<unsigned int>(RAMFS_MAGIC):
       return true;
   }
   return false;
